### PR TITLE
fix bug with rmi multiple tag

### LINKED ISSUE #11414: Docker Image Removal Bug By Id
There is a bug with image removal where an image with multiple tags will be allowed to be removed regardless of whether that image is in use or not if you try to remove by id.

Steps to Reproduce:

    # docker pull drolnick/spin:v2
    Pulling repository drolnick/spin
    38b8f2f5dc2b: Download complete 
    511136ea3c5a: Download complete 
    36669626e49c: Download complete 
    d5570ef1464a: Download complete 
    b3936f269611: Download complete 
    Status: Downloaded newer image for drolnick/spin:v2

    # docker run -d drolnick/spin:v2
    800851c61645b15161e3631dac59db6fa2dc98de9d43e3ac17c688ef7b7193d6

    # docker rmi 38b8f2f5dc2b
    Untagged: drolnick/spin:v2
    Untagged: drolnick/spin:v2_1
    Untagged: drolnick/spin:v2_3
    Deleted: 38b8f2f5dc2b34f428f778abca5c982ec8b36d0e4c63d3e943882eea2dac1946
    Deleted: b3936f26961179f0fe7e217c13236497af012980139c307d6402d9d4d6c5d50a
    Deleted: d5570ef1464a43fe282dd2705b38a2d739812b0b8036a49cfa09811737cfbed4
    Deleted: 36669626e49c623f3b2aea41052bbc9de0506807d92137bf76c0fd3264eaff34
    Deleted: 511136ea3c5a64f264b78b5433614aec563103b4d4702f3ba7d4d2698e22c158

    # docker images
    REPOSITORY          TAG                 IMAGE ID            CREATED             VIRTUAL SIZE

Compare that to the effect that happens when you do the following

    # docker pull drolnick/spin:v2
    Pulling repository drolnick/spin
    38b8f2f5dc2b: Download complete 
    511136ea3c5a: Download complete 
    36669626e49c: Download complete 
    d5570ef1464a: Download complete 
    b3936f269611: Download complete 
    Status: Downloaded newer image for drolnick/spin:v2

    # docker run -d drolnick/spin:v2
    800851c61645b15161e3631dac59db6fa2dc98de9d43e3ac17c688ef7b7193d6

    # docker rmi drolnick/spin:v2
    Untagged: drolnick/spin:v2

    # docker rmi drolnick/spin:v2_1
    Untagged: drolnick/spin:v2_1

    # docker rmi drolnick/spin:v2_3
    Error response from daemon: Conflict, cannot delete 38b8f2f5dc2b because the running container 800851c61645 is using it, stop it and use -f to force
    FATA[0000] Error: failed to remove one or more images 

    # docker images
    REPOSITORY          TAG                 IMAGE ID            CREATED             VIRTUAL SIZE
    drolnick/spin       v2_3                38b8f2f5dc2b        About an hour ago   84.98 MB

We tracked this down to **docker/daemon/image_delete.go**  The function DeleteImage only checks  canDeleteImage if there is only a single tag mapping to an image.  

    # docker version
    Client version: 1.5.0
    Client API version: 1.17
    Go version (client): go1.4.1
    Git commit (client): a8a31ef
    OS/Arch (client): linux/ad64
    Server version: 1.5.0
    Server API version: 1.17
    Go version (server): go1.4.1
    Git commit (server): a8a31ef

### PR DESCRIPTION
**Pull request by jessfraz**
fixes #11414 once and for all

this code desperately needs to be refactored, but for now here is the fix and a new test, I will refactor after ;)

### CODE REVIEW

File: integration-cli/docker_cli_rmi_test.go [hunk 1]
```diff
@@ -74,6 +74,56 @@ func (s *DockerSuite) TestRmiTag(c *check.C) {
...
+	// first checkout without force it fails
+	out, _, err = runCommandWithOutput(exec.Command(dockerBinary, "rmi", imgID))
+	expected := fmt.Sprintf("Conflict, cannot delete %s because the running container %s is using it, stop it and use -f to force", imgID[:12], containerID[:12])
+	if err == nil || !strings.Contains(out, expected) {
+		c.Fatalf("rmi tagged in multiple repos should have failed without force: %s, %v, expected: %s", out, err, expected)
+	}
+
+	dockerCmd(c, "stop", containerID)
+	dockerCmd(c, "rmi", "-f", imgID)
+	{
```
Comment (cpuguy83) [inline integration-cli/docker_cli_rmi_test.go]: What's up with these braces?

**Comment by cpuguy83** [comment]
LGTM if janky is good.

File: integration-cli/docker_cli_rmi_test.go (hunk 1 above)
Comment (jfrazelle) [inline integration-cli/docker_cli_rmi_test.go]: copy paste from test below ;)

Comment (jfrazelle) [inline integration-cli/docker_cli_rmi_test.go]: i will remove

Comment (jfrazelle) [inline integration-cli/docker_cli_rmi_test.go]: updated

**Comment by vieux** [comment]
@jimmyxian can you please make sure swarm follows the same behaviour ? Thank you :smile:

**Comment by calavera** [comment]
LGTM

**Comment by thaJeztah** [comment]
:heart:  thanks Jess!

**Comment by jimmyxian** [comment]
@vieux Sure, I will check it  :)

**Comment by jimmyxian** [comment]
@vieux I have checked it. It's OK. swarm can follow this.

**Comment by jfrazelle** [comment]
cherrypicked

**Pull request merged**